test/test.js

```diff
@@ -14,290 +19,476 @@ function log () {
   console.log.apply(console, args)
 }
 
-var testfile = __dirname + '/testfile.txt'
+var testfilePath = __dirname + '/testfile.txt'
+var testfile = fs.readFileSync(__dirname + '/testfile.txt')
+
+describe('IPFS Node.js API wrapper tests', function () {
+  var apiClients = {} // a, b, c
+  var ipfsNodes = {} // a, b, c
 
-describe('ipfs node api', function () {
-  var ipfs, ipfsNode
   before(function (done) {
     this.timeout(20000)
     log('ipfs node setup')
+    var counter = 0
+
+    if (isNode) {
+      startIndependentNode(ipfsNodes, apiClients, 'a', finish)
+      startIndependentNode(ipfsNodes, apiClients, 'b', finish)
+      startIndependentNode(ipfsNodes, apiClients, 'c', finish)
```

Comment:
So, then starting the nodes by gulp scripts as well? What is the best way to pass the disposable Node addrs to the Karma test?

Should the Node.js tests also use Gulp to spawn the disposable Node.js?

Trying to figure out what is more common/clean for these type of situations